### PR TITLE
sdf -> usd: Added Visual and geometries

### PR DESCRIPTION
**Pull request by ahcorde**
Signed-off-by: ahcorde <ahcorde@gmail.com>
# 🎉 New feature

## Summary

This PR adds support for Visuals and Geometries
 - Support for visuals and geometries
 - Visual and Geometry tests
 - Some minor improvements in doc and tests

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by koonpeng** [comment]
When trying to convert the ur10 model, the meshes are missing,

```
        def Xform "base" (
            prepend apiSchemas = ["PhysicsRigidBodyAPI", "PhysicsMassAPI"]
        )
        {
            point3f physics:centerOfMass = (0, 0, 0)
            float3 physics:diagonalInertia = (0.00610633, 0.00610633, 0.01125)
            float physics:mass = 4
            float3 xformOp:rotateXYZ = (0, -0, 0)
            double3 xformOp:translate = (0, 0, 0)
            uniform token[] xformOpOrder = ["xformOp:translate", "xformOp:rotateXYZ"]
        }
```

**Comment by adlarkin** [comment]
> I tried to convert the ur10 model, but it cannot find the mesh files

Do you need to use this PR for material support? https://github.com/ignitionrobotics/sdformat/pull/831

**Comment by koonpeng** [comment]
> Do you need to use this PR for material support? #831

I did not use the PR, but I don't think it is needed. I heard from @ahcorde that the transformations for ur10 downloaded from ignition fuel is wrong so I am trying to reproduce it.

**Review by koonpeng — APPROVED** [review]

File: usd/src/sdf_parser/Geometry.cc [hunk 1]
```diff
@@ -0,0 +1,460 @@
...
+
+      primName = removeDash(primName);
+
+      auto usdMesh = pxr::UsdGeomMesh::Define(_stage, pxr::SdfPath(primName));
+      usdMesh.CreatePointsAttr().Set(meshPoints);
+      usdMesh.CreateFaceVertexIndicesAttr().Set(faceVertexIndices);
+      usdMesh.CreateFaceVertexCountsAttr().Set(faceVertexCounts);
+
+      auto coordinates = usdMesh.CreatePrimvar(
+          pxr::TfToken("st"), pxr::SdfValueTypeNames->Float2Array,
```
Comment (adlarkin) [inline usd/src/sdf_parser/Geometry.cc:365]: Is there a reason why the token here is `st`? This is a PrimVar for UV coordinates, so would it be better to set the token to `uv`?

**Review by adlarkin — COMMENTED** [review]
I made a few updates in https://github.com/ignitionrobotics/sdformat/pull/829/commits/379defde10447d6ab0b4dfdd4f5af1c7f20e320c. Overall, LGTM once the questions I have below are addressed.

File: usd/src/sdf_parser/Geometry.cc (hunk 1 above)
Comment (ahcorde) [reply usd/src/sdf_parser/Geometry.cc:365]: it's just a different naming convention.

uv coordinates start from the upper left corner (v-axis is facing down).
st coordinates start from the lower left corner (t-axis is facing up).
```
s = u;
t = 1-v;
```

File: usd/src/sdf_parser/Geometry.cc [hunk 2]
```diff
@@ -0,0 +1,460 @@
...
+      // array only allows for all elements to have one value, which in
+      // this case is "verticesPerFace"
+      for (unsigned int n = 0; n < numFaces; ++n)
+        faceVertexCounts.push_back(verticesPerFace);
+
+      std::string primName;
+      if (!subMesh->Name().empty())
+        primName = _path + "/" + subMesh->Name();
+      else
+        primName = _path + "/submesh_" + std::to_string(i);
```
[the remark this answers is not in the document]
Comment (ahcorde) [reply usd/src/sdf_parser/Geometry.cc:349]: this is because a mesh should have more than one submesh

**Comment by ahcorde** [comment]
@scpeters or @azeey do you mind to review it ?

File: usd/src/sdf_parser/sdf2usd_TEST.cc [hunk 3]
```diff
@@ -69,7 +69,7 @@ TEST(check_cmd, IGN_UTILS_TEST_DISABLED_ON_WIN32(SDF))
   // Check a good SDF file
   {
     std::string path = ignition::common::joinPaths(pathBase,
-      "shapes_world.sdf");
+      "lights.sdf");
```
[the remark this answers is not in the document]
Comment (ahcorde) [reply usd/src/sdf_parser/sdf2usd_TEST.cc:72]: This file provide more entities to check. There is no particular reason.

**Review by adlarkin — APPROVED** [review]

**Comment by osrf-triage** [comment]
This pull request has been mentioned on **Gazebo Community**. There might be relevant details there:

https://community.gazebosim.org/t/new-ignition-releases-2022-04-13-fortress-edifice/1367/1

**Pull request merged**